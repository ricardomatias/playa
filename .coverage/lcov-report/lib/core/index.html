<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/core</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/core
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.17% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>343/389</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.98% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>141/172</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.82% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>57/68</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.83% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>332/378</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Chord.js"><a href="Chord.js.html">Chord.js</a></td>
	<td data-value="96.9" class="pic high"><div class="chart"><div class="cover-fill" style="width: 96%;"></div><div class="cover-empty" style="width:4%;"></div></div></td>
	<td data-value="96.9" class="pct high">96.9%</td>
	<td data-value="129" class="abs high">125/129</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="44" class="abs high">40/44</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="96.77" class="pct high">96.77%</td>
	<td data-value="124" class="abs high">120/124</td>
	</tr>

<tr>
	<td class="file high" data-value="HarmonyBase.js"><a href="HarmonyBase.js.html">HarmonyBase.js</a></td>
	<td data-value="84.62" class="pic high"><div class="chart"><div class="cover-fill" style="width: 84%;"></div><div class="cover-empty" style="width:16%;"></div></div></td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="84.62" class="pct high">84.62%</td>
	<td data-value="13" class="abs high">11/13</td>
	</tr>

<tr>
	<td class="file high" data-value="Key.js"><a href="Key.js.html">Key.js</a></td>
	<td data-value="94.67" class="pic high"><div class="chart"><div class="cover-fill" style="width: 94%;"></div><div class="cover-empty" style="width:6%;"></div></div></td>
	<td data-value="94.67" class="pct high">94.67%</td>
	<td data-value="75" class="abs high">71/75</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="36" class="abs high">30/36</td>
	<td data-value="92.31" class="pct high">92.31%</td>
	<td data-value="13" class="abs high">12/13</td>
	<td data-value="94.67" class="pct high">94.67%</td>
	<td data-value="75" class="abs high">71/75</td>
	</tr>

<tr>
	<td class="file low" data-value="Mode.js"><a href="Mode.js.html">Mode.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	</tr>

<tr>
	<td class="file high" data-value="Note.js"><a href="Note.js.html">Note.js</a></td>
	<td data-value="89.74" class="pic high"><div class="chart"><div class="cover-fill" style="width: 89%;"></div><div class="cover-empty" style="width:11%;"></div></div></td>
	<td data-value="89.74" class="pct high">89.74%</td>
	<td data-value="78" class="abs high">70/78</td>
	<td data-value="90.48" class="pct high">90.48%</td>
	<td data-value="42" class="abs high">38/42</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="19" class="abs high">16/19</td>
	<td data-value="89.61" class="pct high">89.61%</td>
	<td data-value="77" class="abs high">69/77</td>
	</tr>

<tr>
	<td class="file high" data-value="Scale.js"><a href="Scale.js.html">Scale.js</a></td>
	<td data-value="93.94" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.94" class="pct high">93.94%</td>
	<td data-value="66" class="abs high">62/66</td>
	<td data-value="82.86" class="pct high">82.86%</td>
	<td data-value="35" class="abs high">29/35</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="93.44" class="pct high">93.44%</td>
	<td data-value="61" class="abs high">57/61</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="types.js"><a href="types.js.html">types.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 15:09:45 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
