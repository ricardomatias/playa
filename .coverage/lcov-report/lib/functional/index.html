<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/functional</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/functional
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.38% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>141/151</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80.26% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>61/76</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">95.65% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>22/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.2% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>137/147</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="chord-progression.js"><a href="chord-progression.js.html">chord-progression.js</a></td>
	<td data-value="96.97" class="pic high"><div class="chart"><div class="cover-fill" style="width: 96%;"></div><div class="cover-empty" style="width:4%;"></div></div></td>
	<td data-value="96.97" class="pct high">96.97%</td>
	<td data-value="33" class="abs high">32/33</td>
	<td data-value="76.19" class="pct medium">76.19%</td>
	<td data-value="21" class="abs medium">16/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="96.88" class="pct high">96.88%</td>
	<td data-value="32" class="abs high">31/32</td>
	</tr>

<tr>
	<td class="file low" data-value="chord.js"><a href="chord.js.html">chord.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

<tr>
	<td class="file high" data-value="drums.js"><a href="drums.js.html">drums.js</a></td>
	<td data-value="94.74" class="pic high"><div class="chart"><div class="cover-fill" style="width: 94%;"></div><div class="cover-empty" style="width:6%;"></div></div></td>
	<td data-value="94.74" class="pct high">94.74%</td>
	<td data-value="19" class="abs high">18/19</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="melodies.js"><a href="melodies.js.html">melodies.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="34" class="abs high">34/34</td>
	<td data-value="95.45" class="pct high">95.45%</td>
	<td data-value="22" class="abs high">21/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	</tr>

<tr>
	<td class="file high" data-value="motif.js"><a href="motif.js.html">motif.js</a></td>
	<td data-value="83.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 83%;"></div><div class="cover-empty" style="width:17%;"></div></div></td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

<tr>
	<td class="file high" data-value="movement.js"><a href="movement.js.html">movement.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="48" class="abs high">48/48</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	</tr>

<tr>
	<td class="file high" data-value="scale.js"><a href="scale.js.html">scale.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 15:09:45 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
