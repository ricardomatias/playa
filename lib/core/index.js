/**
* @namespace {module} Core
*/
export { Note } from './Note';
export { default as Chord } from './Chord';
export { default as Scale } from './Scale';
export { default as Key } from './Key';
export { Time } from './Time';
export { Phrase } from './Phrase';
export { PianoRoll } from './PianoRoll';

