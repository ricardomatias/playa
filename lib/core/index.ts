export * from './Note';
export * from './Chord';
export * from './Scale';
export * from './Key';
export * from './Time';

export * from './Event';
export * from './ChordEvent';
export * from './NoteEvent';
