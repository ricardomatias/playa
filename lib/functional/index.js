export { default as chord } from './chord';
export { default as motif } from './motif';
