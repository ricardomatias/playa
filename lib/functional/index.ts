/**
 * Functional
 * @namespace {module} Functional
 */
export { createMotif } from './motif';
export { createChordProgression } from './chord-progression';
export { createMelodies } from './melodies';
export { createPercussion } from './percussion';
export { createArp } from './arp';

export * from './movement';
