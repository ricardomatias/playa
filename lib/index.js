import * as Constants from './constants';
import * as Core from './core';
import * as Functional from './functional';
import * as Tools from './tools';

export {
	Core,
	Constants,
	Functional,
	Tools,
};
