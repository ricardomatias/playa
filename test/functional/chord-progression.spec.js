import { Key, Chord } from '../../lib/core';
import { createMovement, createChordProgression } from '../../lib/functional';
import Random from '../../lib/tools/random';

describe('A Chord Progression test suite', () => {
	it('should generate chords based on a movement timeline - free rhythm', () => {
		// given
		Random.setSeed('test');

		const aMaj = new Key('A', Key.MAJOR);

		const opts = {
			modProb: 0.40,
		};

		// when
		const movement = createMovement(aMaj, '4:0:0', 6, opts);

		const prog = createChordProgression(movement.timeline);

		expect(prog).toMatchSnapshot();
	});

	it('should generate chords based on a movement timeline - turn rhythm', () => {
		// given
		Random.setSeed('test');

		const aMaj = new Key('A', Key.MAJOR);

		const opts = {
			timeSignatures: [ [ 4, 4 ] ],
			modProb: 0.40,
		};

		// when
		const movement = createMovement(aMaj, '4:0:0', 6, opts);

		const prog = createChordProgression(movement.timeline, { rhythmType: 'turn', restProb: 0.3 });

		expect(prog).toMatchSnapshot();
	});

	it('should generate progression with defined structures', () => {
		// given
		Random.setSeed('test-2');

		const aMaj = new Key('Db', Key.MAJOR);

		const opts = {
			modProb: 0.40,
		};

		// when
		const movement = createMovement(aMaj, '4:0:0', 8, opts);

		const prog = createChordProgression(movement.timeline, {
			structures: [ Chord.SIXTH ],
			inversionProb: 0.5,
			octaves: [ [ 4, 1 ], [ 2, 2 ] ],
			rhythms: [ '4nt', '4n' ],
			minChordNotes: 2,
		});

		expect(prog).toMatchSnapshot();
	});
});
