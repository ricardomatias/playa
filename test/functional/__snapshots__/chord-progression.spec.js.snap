// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Chord Progression test suite should generate chords based on a movement timeline - free rhythm 1`] = `
Array [
  Object {
    "chord": Array [
      66,
      69,
      73,
    ],
    "chordName": "F#m",
    "dur": 960,
    "time": 0,
  },
  Object {
    "chord": Array [
      66,
      69,
      73,
    ],
    "chordName": "F#m",
    "dur": 480,
    "time": 960,
  },
  Object {
    "chord": Array [
      66,
      71,
      73,
    ],
    "chordName": "F#sus4",
    "dur": 960,
    "time": 1440,
  },
  Object {
    "chord": Array [
      66,
      71,
      73,
    ],
    "chordName": "F#sus4",
    "dur": 480,
    "time": 2400,
  },
  Object {
    "chord": Array [
      64,
      68,
      73,
    ],
    "chordName": "C#m",
    "dur": 960,
    "time": 2880,
  },
  Object {
    "chord": Array [
      64,
      66,
      69,
      73,
    ],
    "chordName": "F#m7",
    "dur": 480,
    "time": 3840,
  },
  Object {
    "chord": Array [
      64,
      66,
      69,
      73,
    ],
    "chordName": "F#m7",
    "dur": 960,
    "time": 4320,
  },
  Object {
    "chord": Array [
      69,
      79,
      83,
      86,
      76,
      72,
    ],
    "chordName": "Am11",
    "dur": 640,
    "time": 5280,
  },
  Object {
    "chord": Array [
      69,
      79,
      83,
      86,
      76,
      72,
    ],
    "chordName": "Am11",
    "dur": 320,
    "time": 5920,
  },
  Object {
    "chord": Array [
      69,
      79,
      83,
      86,
      76,
      72,
    ],
    "chordName": "Am11",
    "dur": 480,
    "time": 6240,
  },
  Object {
    "chord": Array [
      69,
      76,
      86,
    ],
    "chordName": "Dsus2",
    "dur": 480,
    "time": 6720,
  },
  Object {
    "chord": Array [
      69,
      76,
      86,
    ],
    "chordName": "Dsus2",
    "dur": 480,
    "time": 7200,
  },
]
`;

exports[`A Chord Progression test suite should generate chords based on a movement timeline - turn rhythm 1`] = `
Array [
  Object {
    "chord": Array [],
    "chordName": "",
    "dur": 240,
    "time": 0,
  },
  Object {
    "chord": Array [
      66,
      69,
      73,
    ],
    "chordName": "F#m",
    "dur": 240,
    "time": 240,
  },
  Object {
    "chord": Array [],
    "chordName": "",
    "dur": 240,
    "time": 0,
  },
  Object {
    "chord": Array [
      66,
      69,
      73,
    ],
    "chordName": "F#m",
    "dur": 720,
    "time": 720,
  },
  Object {
    "chord": Array [
      66,
      71,
      73,
    ],
    "chordName": "F#sus4",
    "dur": 320,
    "time": 1440,
  },
  Object {
    "chord": Array [
      66,
      71,
      73,
    ],
    "chordName": "F#sus4",
    "dur": 480,
    "time": 1760,
  },
  Object {
    "chord": Array [],
    "chordName": "",
    "dur": 320,
    "time": 1440,
  },
  Object {
    "chord": Array [
      66,
      71,
      73,
    ],
    "chordName": "F#sus4",
    "dur": 320,
    "time": 2560,
  },
  Object {
    "chord": Array [
      61,
      64,
      68,
      71,
    ],
    "chordName": "C#m7",
    "dur": 240,
    "time": 2880,
  },
  Object {
    "chord": Array [
      61,
      64,
      68,
      71,
    ],
    "chordName": "C#m7",
    "dur": 240,
    "time": 3120,
  },
  Object {
    "chord": Array [],
    "chordName": "",
    "dur": 480,
    "time": 2880,
  },
  Object {
    "chord": Array [
      66,
      69,
      73,
      76,
    ],
    "chordName": "F#m7",
    "dur": 320,
    "time": 3840,
  },
  Object {
    "chord": Array [
      66,
      69,
      73,
      76,
    ],
    "chordName": "F#m7",
    "dur": 480,
    "time": 4160,
  },
  Object {
    "chord": Array [
      66,
      69,
      73,
      76,
    ],
    "chordName": "F#m7",
    "dur": 320,
    "time": 4640,
  },
  Object {
    "chord": Array [],
    "chordName": "",
    "dur": 320,
    "time": 3840,
  },
  Object {
    "chord": Array [
      64,
      69,
      74,
    ],
    "chordName": "Asus4",
    "dur": 320,
    "time": 5280,
  },
  Object {
    "chord": Array [
      64,
      69,
      74,
    ],
    "chordName": "Asus4",
    "dur": 480,
    "time": 5600,
  },
  Object {
    "chord": Array [
      64,
      69,
      74,
    ],
    "chordName": "Asus4",
    "dur": 480,
    "time": 6080,
  },
  Object {
    "chord": Array [
      64,
      69,
      74,
    ],
    "chordName": "Asus4",
    "dur": 160,
    "time": 6560,
  },
  Object {
    "chord": Array [],
    "chordName": "",
    "dur": 240,
    "time": 6720,
  },
  Object {
    "chord": Array [
      62,
      67,
      69,
    ],
    "chordName": "Dsus4",
    "dur": 240,
    "time": 6960,
  },
  Object {
    "chord": Array [],
    "chordName": "",
    "dur": 480,
    "time": 6720,
  },
]
`;

exports[`A Chord Progression test suite should generate progression with defined structures 1`] = `
Array [
  Object {
    "chord": Array [
      53,
      56,
      60,
      61,
      65,
      68,
      72,
      73,
    ],
    "chordName": "Fm6m",
    "dur": 320,
    "time": 0,
  },
  Object {
    "chord": Array [
      53,
      56,
      60,
      61,
      65,
      68,
      72,
      73,
    ],
    "chordName": "Fm6m",
    "dur": 320,
    "time": 320,
  },
  Object {
    "chord": Array [
      53,
      56,
      60,
      61,
      65,
      68,
      72,
      73,
    ],
    "chordName": "Fm6m",
    "dur": 320,
    "time": 640,
  },
  Object {
    "chord": Array [
      54,
      58,
      61,
      63,
    ],
    "chordName": "Gb6",
    "dur": 320,
    "time": 960,
  },
  Object {
    "chord": Array [
      54,
      58,
      61,
      63,
    ],
    "chordName": "Gb6",
    "dur": 320,
    "time": 1280,
  },
  Object {
    "chord": Array [
      54,
      58,
      61,
      63,
    ],
    "chordName": "Gb6",
    "dur": 320,
    "time": 1600,
  },
  Object {
    "chord": Array [
      54,
      58,
      61,
      65,
    ],
    "chordName": "Bbm6m",
    "dur": 320,
    "time": 1920,
  },
  Object {
    "chord": Array [
      54,
      58,
      61,
      65,
    ],
    "chordName": "Bbm6m",
    "dur": 320,
    "time": 2240,
  },
  Object {
    "chord": Array [
      54,
      58,
      61,
      65,
    ],
    "chordName": "Bbm6m",
    "dur": 320,
    "time": 2560,
  },
  Object {
    "chord": Array [
      53,
      58,
      61,
      68,
    ],
    "chordName": "Db6",
    "dur": 320,
    "time": 2880,
  },
  Object {
    "chord": Array [
      53,
      58,
      61,
      68,
    ],
    "chordName": "Db6",
    "dur": 320,
    "time": 3200,
  },
  Object {
    "chord": Array [
      53,
      58,
      61,
      68,
    ],
    "chordName": "Db6",
    "dur": 320,
    "time": 3520,
  },
  Object {
    "chord": Array [
      53,
      60,
      65,
    ],
    "chordName": "Fm6m",
    "dur": 320,
    "time": 3840,
  },
  Object {
    "chord": Array [
      53,
      60,
      65,
    ],
    "chordName": "Fm6m",
    "dur": 320,
    "time": 4160,
  },
  Object {
    "chord": Array [
      53,
      60,
      65,
    ],
    "chordName": "Fm6m",
    "dur": 320,
    "time": 4480,
  },
  Object {
    "chord": Array [
      53,
      61,
      66,
      70,
    ],
    "chordName": "Bbm6m",
    "dur": 320,
    "time": 4800,
  },
  Object {
    "chord": Array [
      53,
      61,
      66,
      70,
    ],
    "chordName": "Bbm6m",
    "dur": 320,
    "time": 5120,
  },
  Object {
    "chord": Array [
      53,
      61,
      66,
      70,
    ],
    "chordName": "Bbm6m",
    "dur": 320,
    "time": 5440,
  },
  Object {
    "chord": Array [
      54,
      63,
      70,
      71,
    ],
    "chordName": "Ebm6m",
    "dur": 480,
    "time": 5760,
  },
  Object {
    "chord": Array [
      54,
      63,
      70,
      71,
    ],
    "chordName": "Ebm6m",
    "dur": 480,
    "time": 6240,
  },
  Object {
    "chord": Array [
      52,
      61,
      68,
      69,
    ],
    "chordName": "Dbm6m",
    "dur": 320,
    "time": 6720,
  },
  Object {
    "chord": Array [
      52,
      61,
      68,
      69,
    ],
    "chordName": "Dbm6m",
    "dur": 320,
    "time": 7040,
  },
  Object {
    "chord": Array [
      52,
      61,
      68,
      69,
    ],
    "chordName": "Dbm6m",
    "dur": 320,
    "time": 7360,
  },
]
`;
